### PR TITLE
Audio: Move components related config to subfolder

### PR DESCRIPTION
**Pull request by andrula-song**
Create a Kconfig for each module and include them from src/audio/Konfig.

### CODE REVIEW

**Comment by btian1** [comment]
@marc-hb , @kv2019i  if each module need a separate Kconfig, we may need add it together to avoid missed modules.

**Review by btian1 — APPROVED** [review]

**Comment by andrula-song** [comment]
SOFCI TEST

File: src/audio/Kconfig [hunk 1]
```diff
@@ -11,18 +12,13 @@ config COMP_BASEFW_IPC4
...
-        help
-          Select for COPIER component
+rsource "copier/Kconfig"
 
 config HOST_DMA_RELOAD_DELAY_ENABLE
 	bool "Delay reloading DMA for host interfaces"
 	default y
 	help
-	  When deep buffers are used the firmware doesn't need to have the
+	  When deep buffers are used the firmware does not need to have the
```
Comment (btian1) [inline src/audio/Kconfig:21]: why make this change?

**Review by RanderWang — APPROVED** [review]

File: src/audio/Kconfig (hunk 1 above)
Comment (andrula-song) [reply src/audio/Kconfig:21]: because the" ' " makes code displays very strange.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Failures known and unrelated.

**Pull request merged**